ReleaseProcessScript/tools/Core/semver_functions.ps1

```diff
@@ -103,11 +103,12 @@ function Get-Possible-Versions-Hotfix ($Version, $MeantForCurrentVersion = $fals
     $NextPossibleFullVersion = "$($Major).$($Minor).$($Patch)"  
 
     $NextPreVersion = [string](1 + $PreVersion)
-    $NextPossiblePreVersion = "$($Major).$($Minor).$($Patch)-$($Pre).$($NextPreVersion)" 
+    $NextPrePatch = $Match.Groups["patch"].ToString()
```

Comment:
I think it would be easier to understand if `$Patch` of line 90 would be immutable and a new variable is created for the possibly mutated patch number on lines 92-96. This way you could just use the initial `$Patch` here without extracting it from the match a second time.

Reply:
I'm with @iam-mholle on this one. The mix of mutable and mixed naming in this method is not helpful towards maintainablity. I tried to come up with a compromise, but there just isn't one. 

Each variable should be assigned only once and b) have a clear name. It will be okay if we duplicate the building of the version number at this time since there's less business logic in the format *) than in using the correct number-parts in the first place.

*) Actually, the version format is completely fixed and easily reviewed so there's no worry this could break in the future. In theory, we would also have a version builder and then only do the formatting once, but that's nice to have, nothing more.

Reply:
I have refactored the method such that it (hopefully) has a better naming scheme and the initially extracted version part variables (from the regex) are not mutated.